miner/worker.go

```diff
@@ -873,71 +885,56 @@ func (w *worker) commitTransaction(tx *types.Transaction, coinbase common.Addres
 			rcfg.ScalarSlot:    {},
 		})
 	for addr, keys := range proofStorages {
-		if _, existed := txStorageTrace.StorageProofs[addr.String()]; !existed {
-			txStorageTrace.StorageProofs[addr.String()] = make(map[string][]hexutil.Bytes)
+		addrStr := addr.String()
+		txStorageTrace.StorageProofs[addrStr] = make(map[string][]hexutil.Bytes)
+		proofCache, existed := proofCaches[addrStr]
+		if !existed {
+			panic("any storage proof under an account must come along with account proof")
 		}
 
-		trie, err := w.current.state.GetStorageTrieForProof(addr)
-		if err != nil {
-			// but we still continue to next address
-			log.Error("Storage trie not available", "error", err, "address", addr)
+		if proofCache.storageTrie == nil {
+			// we have no storage proof available (maybe the account is not existed yet)
+			// , just continue to next address
+			log.Info("Storage trie not available", "address", addr)
 			continue
 		}
-		zktrieTracer := w.current.state.NewProofTracer(trie)
 
 		for key, values := range keys {
-			addrStr := addr.String()
+
 			keyStr := key.String()
-			isDelete := bytes.Equal(values.Bytes(), common.Hash{}.Bytes())
 
-			m, existed := txStorageTrace.StorageProofs[addrStr]
+			stgProof, existed := proofCache.storageProof[keyStr]
 			if !existed {
```

Comment:
don't we need to `proofCache.trieTracer.MarkDeletion(key)` (for `isDelete `) even if `existed` is `true`?
(The original `still need to touch tracer for deletion` logic in `eth/tracers/api_blocktrace.go`)

Reply:
You are right. I will fix that

Reply:
fixed